module/VuFind/tests/integration-tests/src/VuFindTest/Mink/EmailAuthenticationTest.php

```diff
@@ -0,0 +1,244 @@
+    {
+        static::failIfDataExists();
+    }
+
+    /**
+     * Test the (non-ILS) email authentication process.
+     *
+     * @return void
+     */
+    public function testEmailAuthentication(): void
+    {
+        $this->setUpDatabaseEmailConfig();
+
+        $this->resetEmailLog();
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Create account
+        $this->clickCss($page, '#loginOptions a');
+        $this->clickCss($page, '.modal-body .createAccountLink');
+        $this->fillInAccountForm($page);
+        $this->clickCss($page, '.modal-body .btn.btn-primary');
+
+        // Log out
+        $this->clickCss($page, '.logoutOptions a.logout');
+
+        // Request login:
+        $this->clickCss($page, '#loginOptions a');
+        $this->findCssAndSetValue($page, '.modal-body #login_Email_username', 'username1@ignore.com');
+        $this->clickCss($page, '.modal-body .btn.btn-primary', null, 1);
+        $this->assertEquals(
+            'We have sent a login link to your email address. It may take a few moments for the link to arrive.'
+            . " If you don't receive the link shortly, please check also your spam filter.",
+            $this->findCssAndGetText($page, '.alert-success')
+        );
+
+        // Extract the link from the provided message:
+        $email = $this->getLoggedEmail();
+        $headers = $email->getHeaders();
+        $body = $email->getBody()->getBody();
+        $this->assertEquals('From: noreply@vufind.org', $headers->get('from')->toString());
+        $this->assertEquals('To: username1@ignore.com', $headers->get('to')->toString());
+        preg_match('/Link to login: <(http.*)>/', $body, $matches);
+        $loginLink = $matches[1];
+
+        // Follow the verification link:
+        $session->visit($loginLink);
+
+        // Log out (we can't log out unless we successfully logged in):
+        $this->clickCss($page, '.logoutOptions a.logout');
+
+        // Clean up the email log:
+        $this->resetEmailLog();
+    }
+
+    /**
+     * Test the (non-ILS) email authentication process with invalid email address.
+     *
+     * @return void
+     *
+     * @depends testEmailAuthentication
+     */
+    public function testEmailAuthenticationBadEmail(): void
+    {
+        $this->setUpDatabaseEmailConfig();
+
+        $this->resetEmailLog();
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Request login:
+        $this->clickCss($page, '#loginOptions a');
+        $this->findCssAndSetValue($page, '.modal-body #login_Email_username', 'username1@foo.bar');
+        $this->clickCss($page, '.modal-body .btn.btn-primary', null, 1);
+        $this->assertEquals(
+            'We have sent a login link to your email address. It may take a few moments for the link to arrive.'
```

Comment:
Would it be better if we informed the user that they provided an invalid email address, rather than acting like things were successful? Or am I misunderstanding the context here?

Reply:
No, we don't want to disclose any information to a potential perpetrator. Perhaps the message could be formatted in a different way, but the main point is to not tell if the email address was found or not.

Reply:
Maybe I misunderstood -- I of course agree that we don't want to disclose whether or not a valid email exists in the database, but I thought this case was to account for the user typing something that isn't an email address at all, in which case a separate message would not disclose anything sensitive and might help somebody who made a typo.

Reply:
Right, of course. In that regard we should do a better job. Perhaps making the input field `type="email"` would make sense too. But I think it's out of scope here.